invenio_search/ext.py

```diff
@@ -246,48 +247,128 @@ def active_aliases(self):
+                    body=json.load(body),
+                    ignore=ignore,
+                ) if not dry_run else None
+            )
+            if create_alias:
+                alias_result = (
+                    final_alias,
+                    self.client.indices.put_alias(
+                        index=final_index,
+                        name=final_alias,
+                        ignore=ignore,
+                    ) if not dry_run else None
+                )
+        return index_result, alias_result
+
     def create(self, ignore=None):
         """Yield tuple with created index name and responses from a client."""
         ignore = ignore or []
+        created_indices = {}
 
-        def _create(tree_or_filename, alias=None):
-            """Create indices and aliases by walking DFS."""
-            # Iterate over aliases:
+        def _build(tree_or_filename, alias=None):
+            """Create indices and aliases by walking DFS.
+
+            Yields the result from create_index() and boolean is_index.
+            """
             for name, value in tree_or_filename.items():
                 if isinstance(value, dict):
-                    for result in _create(value, alias=name):
+                    for result in _build(value, alias=name):
                         yield result
                 else:
-                    with open(value, 'r') as body:
-                        yield name, self.client.indices.create(
-                            index=name,
-                            body=json.load(body),
+                    index_result, alias_result = \
+                        self.create_index(
+                            name,
                             ignore=ignore,
+                            dry_run=True
                         )
-
+                    if index_result[0]:
+                        created_indices[name] = index_result[0]
+                        yield index_result, name, True
+                    if alias_result[0]:
+                        data = dict(
+                            index=index_result[0],
+                            name=alias_result[0]
+                        )
+                        yield alias_result, data, False
             if alias:
-                yield alias, self.client.indices.put_alias(
-                    index=list(_get_indices(tree_or_filename)),
-                    name=alias,
-                    ignore=ignore,
+                alias_indices = self._get_indices(tree_or_filename)
+                alias_indices = [created_indices[i] for i in alias_indices]
+                alias_name = build_index_name(alias, suffix='', app=self.app)
+                data = dict(
+                    index=alias_indices,
+                    name=alias_name
                 )
+                yield (alias_name, None), data, False
 
-        for result in _create(self.active_aliases):
-            yield result
+        results = list(_build(self.active_aliases))
```

Comment:
Since we've gone with this "actions-tracking" method, you can build more flexible/comfortable objects in inside the `_build` and instead of yielding then you add them to a dictionary process them later below. Let's discuss IRL though about it